ang/civicase/DashboardTab.js

```diff
@@ -47,39 +75,79 @@
     }
 
     /**
-     * Returns a copy of the default query params for cases, interpolated with
-     * the currently selected relationship type filter's params
+     * Get the processed list of query pars of the given collection (cases, milestones, etc)
      *
-     * @return {Object}
+     * @param {String}
+     * @return {Object{}}
+     */
+    function getQueryParams (collection) {
+      var activityFiltersCopy = _.cloneDeep($scope.activityFilters);
+
+      // Cases need the properties of `activityFilters.case_filter` in the
+      // object root
+      return _.assign({}, defaultsMap[collection], (collection === 'cases'
+        ? activityFiltersCopy.case_filter
+        : activityFiltersCopy
+      ));
+    }
+
+    /**
+     * Initializes the controller watchers
      */
-    function casesQueryParams () {
-      return _.assign({}, CASES_QUERY_PARAMS_DEFAULTS, $scope.activityFilters.case_filter);
+    function initWatchers () {
+      $scope.$watchCollection('filters.caseRelationshipType', function (newType, oldType) {
+        if (newType !== oldType) {
+          $scope.newCasesPanel.query.params = getQueryParams('cases');
+          $scope.newMilestonesPanel.query.params = getQueryParams('milestones');
+        }
+      });
+
+      // When the involvement filters change, broadcast the event that will be
+      // caught by the activity-filters-contact directive which will add the
+      // correct query params to match the filter value
+      $scope.$watch('newMilestonesPanel.custom.involvementFilter', function (newValue, oldValue) {
+        if (newValue === oldValue) {
+          return;
+        }
+
+        $rootScope.$broadcast(
+          'civicaseActivityFeed.query',
+          $scope.newMilestonesPanel.custom.involvementFilter,
+          $scope.newMilestonesPanel.query.params,
+          true
+        );
+      }, true);
     }
 
     /**
-     * Sets the range of the case start date
+     * Sets the range of the date of the entity (Case / Activity)
      *
+     * @param {String} property the property where the information about the date is stored
+     * @param {String} format the date format
+     * @param {Boolean} useNowAsStart whether the starting point should be the current datetime
      * @param {String} selectedRange the currently selected period range
      * @param {Object} queryParams
      */
-    function casesRangeHandler (selectedRange, queryParams) {
+    function rangeHandler (property, format, useNowAsStart, selectedRange, queryParams) {
```

Comment:
Can we simplify the parameters in this function? I think using the curry function makes this more complicated and hides the fact where all these parameters come from. I'd consider returning a plain old handler function:

```js
function getHandlerForRangeChange(fieldName, options) {
  // options contain the format and useNowAsStart

  return function rangeChangeHandler (selectedRange, queryParams) {
    // ...
  }
}
```